### ISSUE TITLE
NoSuchElementException from LS

### ISSUE THREAD

**Issue by prabushi** [ballerina-platform/ballerina-lang#28171]
**Description:**
Getting the following error, and fails for completions and diagnostics.

```
[Error - 12:12:33 PM] Operation 'text/completion' failed! {uri: '/Users/prabushi/Desktop/workspace/single-file/main.bal', [14:15], error: 'No value present'}
java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.get(Optional.java:148)
	at org.ballerinalang.langserver.completions.providers.AbstractCompletionProvider.lambda$getModuleCompletionItems$2(AbstractCompletionProvider.java:256)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.ballerinalang.langserver.completions.providers.AbstractCompletionProvider.getModuleCompletionItems(AbstractCompletionProvider.java:267)
	at org.ballerinalang.langserver.completions.providers.context.BlockNodeContextProvider.getCompletions(BlockNodeContextProvider.java:96)
	at org.ballerinalang.langserver.completions.providers.context.FunctionBodyBlockNodeContext.getCompletions(FunctionBodyBlockNodeContext.java:45)
	at org.ballerinalang.langserver.completions.providers.context.FunctionBodyBlockNodeContext.getCompletions(FunctionBodyBlockNodeContext.java:36)
	at org.ballerinalang.langserver.completions.util.CompletionUtil.route(CompletionUtil.java:99)
	at org.ballerinalang.langserver.completions.util.CompletionUtil.getCompletionItems(CompletionUtil.java:56)
	at org.ballerinalang.langserver.completions.BallerinaCompletionExtension.execute(BallerinaCompletionExtension.java:51)
	at org.ballerinalang.langserver.completions.BallerinaCompletionExtension.execute(BallerinaCompletionExtension.java:37)
	at org.ballerinalang.langserver.LangExtensionDelegator.completion(LangExtensionDelegator.java:92)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$completion$0(BallerinaTextDocumentService.java:134)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

```

**Steps to reproduce:**
```
import ballerina/lang.'query;
import ballerina/io;

const string GET = "GET";
const POST = "POST";
const int I = 125;

type ACTION GET|POST;

public function main() {
    boolean cc = true;
    boolean aa = false;
    boolean bb = true;
    io:print(o);
}

function getValue(string key) returns string|error {
    if (key == "") {
        error err = error("key '" + key + "' not found");
        return err;
    } else {
        return "this is a value";
    }
}

```

Edit and add an non-existing value eg: `io:print(o);` -> `io:print(q);`

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by hasithaa** [comment]
@nadeeshaan Can you refer to the PR and close this.

**Comment by nadeeshaan** [comment]
Fixed with #28405. Hence closing the issue
